packages/stream-metadata/src/routes/spaceMetadata.ts

```diff
@@ -1,36 +1,138 @@
-import { FastifyRequest, FastifyReply } from 'fastify'
+import { FastifyRequest, FastifyReply, FastifyBaseLogger } from 'fastify'
+import { SpaceInfo } from '@river-build/web3'
+import { makeStreamId, StreamPrefix, StreamStateView } from '@river-build/sdk'
+import { ChunkedMedia } from '@river-build/proto'
 
 import { isValidEthereumAddress } from '../validators'
+import { getSpaceDapp } from '../contract-utils'
+import { config } from '../environment'
+import { getStream } from '../riverStreamRpcClient'
+import { getSpaceImage } from './spaceImage'
 
-export function fetchSpaceMetadata(
+export interface SpaceMetadataResponse {
+	name: string
+	description: string
+	image: string | undefined
+}
+
+export async function fetchSpaceMetadata(
 	request: FastifyRequest,
 	reply: FastifyReply,
 	serverUrl: string,
 ) {
 	const logger = request.log.child({ name: fetchSpaceMetadata.name })
 	const { spaceAddress } = request.params as { spaceAddress?: string }
-	logger.info({ spaceAddress }, 'GET /space/../metadata')
 
 	if (!spaceAddress) {
+		logger.error('spaceAddress parameter is required')
```

Comment:
@tak-hntlabs this is perfectly fine and expected from the service's perspective, so it should be `.warn` or `.info` even. it would be fine to log an error on the clients that interface with this endpoint.

Reply:
fixed